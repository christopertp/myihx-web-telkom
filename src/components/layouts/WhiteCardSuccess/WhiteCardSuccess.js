import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './styles.scoped.css';

export default function WhiteCardSuccess(props) {
  const { children, icon, subtitle, title } = props;

  useEffect(() => {
    const body = document.getElementsByTagName('body')[0];
    body.className = 'white-card';
    return () => {
      body.className = '';
    };
  }, []);

  return (
    <>
      <nav className={styles.nav}>
        <Link to="/">
          <img alt="Indihome Logo" src="/assets/logo_indihome-dark.png" />
        </Link>
      </nav>
      <main className={styles.main}>
        <img alt={icon} src={`/assets/grfx_${icon}.svg`} />
        <div className={styles.title}>
          <h1>{title}</h1>
          <p>{subtitle}</p>
        </div>
        {children}
      </main>
    </>
  );
}

WhiteCardSuccess.defaultProps = {
  children: null,
  icon: '',
  subtitle: '',
  title: '',
};

WhiteCardSuccess.propTypes = {
  children: PropTypes.node,
  icon: PropTypes.string,
  subtitle: PropTypes.string,
  title: PropTypes.string,
};
