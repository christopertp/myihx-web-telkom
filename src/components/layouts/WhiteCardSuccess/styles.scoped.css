.main {
  background-color: #ffffff;
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  margin: 0 auto 3.75rem;
  padding: 1.5rem;
  width: 31.875rem;
}

.main > img {
  display: block;
  margin: 0 auto 1rem;
  width: 12.5rem;
}

.nav {
  background-color: #ffffff;
  height: 5.375rem;
  padding: 2rem 0 1.75rem;
  margin-bottom: 2rem;
  text-align: center;
}

.nav > a > img {
  height: 100%;
}

.title {
  margin-bottom: 1.5rem;
}

.title > h1 {
  font-size: 1.25rem;
  font-weight: 800;
  margin: 0 auto 0.5rem;
  max-width: 25rem;
  text-align: center;
}

.title > p {
  line-height: 1.6;
  text-align: center;
  margin: 0 auto;
  max-width: 25rem;
}

@media only screen and (max-width: 48em) {
  .main {
    background-color: transparent;
    border-radius: 0;
    box-shadow: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 0;
    min-height: 100vh;
    padding: 0 1rem;
    width: 100vw;
  }
  
  .main > img {
    width: 11rem;
  }

  .main::after {
    background: url(/assets/bg_header_white_full.png) no-repeat;
    background-color: #ffffff;
    background-size: contain;
    content: '';
    height: 100%;
    left: 0;
    position: fixed;
    top: 35vh;
    width: 100%;
    z-index: -1;
  }

  .main > a {
    position: absolute;
    bottom: 1rem;
    width: calc(100% - 2rem);
  }

  .nav {
    display: none;
  }

  .title > h1 {
    font-size: 1.5rem;
  }
}