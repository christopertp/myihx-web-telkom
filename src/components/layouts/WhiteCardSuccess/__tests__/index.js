import React, { useEffect } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import WhiteCardSuccess from '../WhiteCardSuccess';

describe('src/components/layouts/WhiteCardSuccess', () => {
  test('render', () => {
    useEffect.mockImplementationOnce(fn => fn()());
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<WhiteCardSuccess />);
    expect(tree).toMatchSnapshot();
  });
});
