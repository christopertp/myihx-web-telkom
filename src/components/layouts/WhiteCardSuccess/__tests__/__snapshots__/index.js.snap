// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/layouts/WhiteCardSuccess render 1`] = `
<React.Fragment>
  <nav>
    <Link
      to="/"
    >
      <img
        alt="Indihome Logo"
        src="/assets/logo_indihome-dark.png"
      />
    </Link>
  </nav>
  <main>
    <img
      alt=""
      src="/assets/grfx_.svg"
    />
    <div>
      <h1>
        
      </h1>
      <p>
        
      </p>
    </div>
  </main>
</React.Fragment>
`;
